packages/react/src/runtime/ax.ts

```diff
@@ -41,7 +48,15 @@ export default function ax(classNames: (string | undefined | null | false)[]): s
       continue;
     }
 
-    const groups = cls.split(' ');
+    let groups: string[] = [];
+    if (typeof cls === 'string') {
+      groups = cls.split(' ');
+    } else if (Array.isArray(cls)) {
+      groups = cls; // NOTE: Not really a valid scenario, so we don't deeply recurse or handle this properly
```

Comment:
this is an area in the codebase i haven't looked at in a while, so i'm not too familiar with the runtime stuff 😅

from the comment and the tests, i take it only APIs outside of Compiled (and never Compiled APIs) would ever create a situation where `Array.isArray(cls)` is true right? (e.g. `className = [['_abcd1234', '_efgh1234'], '_ijkl1234 mnop1234', ...]`)

i don't have a preference on whether to keep the `else if (Array.isArray(cls)) { ... }` branch in or not, just aiming to understand the PR fully

Reply:
Just to drop the scenario, right now let's say if you do like this:
```tsx
const Component = (props) => (
  <div className={xcss} css={{ margin: 0 }} />
);

export default (props) => <Component xcss={[{ margin: '12px' }, props.xcss]} />
`````

You will get a type error because our `xcss` implementation requires `xcss={cx({ margin: '12px' }, props.xcss)}` and arrays won't work, but in theory I could see people copy + pasting `css={[…]}` array into `xcss={[…]}` and ignoring it and it throwing an error (because I kind of did that 🤣).

I don't mind much either way to be honest, it does have a type error.

I would prefer to either `throw new Error(…)` in here or keep it, because if it does get passed, doing `String(array).split(' ');` is quite unperformant.

Reply:
that makes sense, thank you for the additional context